### ISSUE TITLE
Helper to identify user portal information

### ISSUE THREAD

**Issue by JosiahParry** [R-ArcGIS/arcgisutils#32]
```r
portals <- arc_base_req(
  "https://arcgis.com/sharing/rest/portals/self",
  arc_token()
) |>
  req_body_form(f = "json") |> 
  req_perform() |> 
  resp_body_string() |> 
  RcppSimdJson::fparse()
```

This function will return a massive list that includes just about every piece of information you could possibly want to know about your user in your authorized portal. 

``` r
library(httr2)
#> Warning: package 'httr2' was built under R version 4.3.1
library(arcgisutils)
#> Warning: package 'arcgisutils' was built under R version 4.3.1
portals <- arc_base_req(
  "https://arcgis.com/sharing/rest/portals/self",
  arc_token()
) |>
  req_body_form(f = "json") |> 
  req_perform() |> 
  resp_body_string() |> 
  RcppSimdJson::fparse()

names(portals)
#>  [1] "2DStylesGroupQuery"                "3DBasemapGalleryGroupQuery"       
#>  [3] "analysisLayersGroupQuery"          "basemapGalleryGroupQuery"         
#>  [5] "cdnUrl"                            "colorSetsGroupQuery"              
#>  [7] "contentCategorySetsGroupQuery"     "customBaseUrl"                    
#>  [9] "defaultBasemap"                    "defaultDevBasemap"                
#> [11] "defaultExtent"                     "defaultVectorBasemap"             
#> [13] "description"                       "devBasemapGalleryGroupQuery"      
#> [15] "featuredGroups"                    "featuredItemsGroupQuery"          
#> [17] "galleryTemplatesGroupQuery"        "helpBase"                         
#> [19] "helperServices"                    "homePageFeaturedContent"          
#> [21] "homePageFeaturedContentCount"      "isPortal"                         
#> [23] "layerTemplatesGroupQuery"          "livingAtlasGroupQuery"            
#> [25] "name"                              "portalHostname"                   
#> [27] "portalMode"                        "portalName"                       
#> [29] "portalThumbnail"                   "rasterFunctionTemplatesGroupQuery"
#> [31] "rotatorPanels"                     "staticImagesUrl"                  
#> [33] "stylesGroupQuery"                  "supports3DTilesServices"          
#> [35] "supportsHostedServices"            "symbolSetsGroupQuery"             
#> [37] "templatesGroupQuery"               "thumbnail"                        
#> [39] "use3dBasemaps"                     "useVectorBasemaps"                
#> [41] "vectorBasemapGalleryGroupQuery"    "httpPort"                         
#> [43] "httpsPort"                         "ipCntryCode"                      
#> [45] "supportsOAuth"                     "isReadOnly"                       
#> [47] "currentVersion"
```

<sup>Created on 2024-03-02 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by JosiahParry** [comment]
For example: 

<img width="651" alt="image" src="https://github.com/R-ArcGIS/arcgisutils/assets/13534508/8a90d4f9-4c9c-4642-bbde-798eff0486d4">

**Comment by JosiahParry** [comment]
This metadata can be included into the token itself after authorization that could make the token kind of big but we would have literally all the information we might ever need...

**Comment by JosiahParry** [comment]
Closed by #36
